src/request/mod.rs

```diff
@@ -43,7 +42,7 @@ pub struct Request<'a, 'b: 'a> {
     pub headers: Headers,
 
     /// The request body as a reader.
-    pub body: Body<'a, 'b>,
+    pub body: Cursor<Vec<u8>>,
```

Comment:
Definitely!  I'm thinking that can just store a future for the body count, either using the hyper Chunk, or streaming out the bytes or some such directly.

That was just the easiest way to make it work/compile now.  I can see about making it nicer for this PR if you wish, or handle it later.  This was just very much geared to be a base to start on :)